### DISCUSSION TITLE
How to create an image for CloudStack CKS with Calico CNI?

### DISCUSSION THREAD

**Discussion by sptest232** [apache/cloudstack#9807] [Q&A] [answered]
Hello!

Help me please. 

There are a couple of images of k8s with Calico CNI https://download.cloudstack.org/cks/
I need to create such images for another version of k8s. 
But when i try to run:

./create-kubernetes-binaries-iso.sh ./ 1.28.5 1.3.0 1.28.0 https://github.com/projectcalico/calico/blob/master/manifests/calico.yaml https://raw.githubusercontent.com/kubernetes/dashboard/v2.7.0/aio/deploy/recommended.yaml setup-v1.28.5

i get an error during the script running:

"ctr: failed to resolve reference "image\"content=\"https://opengraph.githubassets.com/067f4b539d79b3e6316b57fbec0dd6103768b3b1fa38a32fe987ee0ada632101/projectcalico/calico\"/><metaproperty=\"og:image:alt\"content=\"Cloudnativenetworkingandnetworksecurity.Contributetoprojectcalico/calicodevelopmentbycreatinganaccountonGitHub.\"/><metaproperty=\"og:image:width\"content=\"1200\"/><metaproperty=\"og:image:height\"content=\"600\"/><metaproperty=\"og:site_name\"content=\"GitHub\"/><metaproperty=\"og:type\"content=\"object\"/><metaproperty=\"og:title\"content=\"calico/manifests/calico.yamlatmaster·projectcalico/calico\"/><metaproperty=\"og:url\"content=\"https://github.com/projectcalico/calico/blob/master/manifests/calico.yaml\"/><metaproperty=\"og:description\"content=\"Cloudnativenetworkingandnetworksecurity.Contributetoprojectcalico/calicodevelopmentbycreatinganaccountonGitHub.\"/>": invalid reference"

Thank you!

**Comment by kiranchavala** [comment]
@sptest232 you need to have docker installed on the management server

**Comment by sptest232** [comment]
@kiranchavala it is intalled

docker -v
Docker version 24.0.7, build 24.0.7-0ubuntu2~22.04.1

there is no problem if i run:

./create-kubernetes-binaries-iso.sh ./ 1.29.1 1.2.0 1.26.0 https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml https://raw.githubusercontent.com/kubernetes/dashboard/v2.7.0/aio/deploy/recommended.yaml setup-1.29.1

but i need calico cni instead flannel

**Reply by weizhouapache** [reply, marked as answer]
@sptest232 
I think the url should be
https://raw.githubusercontent.com/projectcalico/calico/refs/heads/master/manifests/calico.yaml

**Reply by sptest232** [reply]
Yes! It works!
Thank you!

**Comment by weizhouapache** [comment]
btw, please refer to description of https://github.com/apache/cloudstack/pull/9186
